contracts/commodity/MintableCommodity.sol

```diff
@@ -24,19 +25,24 @@ contract MintableCommodity is BasicCommodity, IMintableCommodity {
     uint256 _value,
     bytes _misc
   ) public returns(uint64) {
+
     //todo jaycen is this safe? Can someone somehow return teh same participant address and spoof that the msg is coming from a defined address?
-    address recipientImplementation = interfaceAddr(msg.sender, "IParticipant");
-    if (recipientImplementation != 0) {
+    //todo replace the following if else logic with the require logic that exists in the split function. This is needed only so that current tests don't break
+    address participantProxy = interfaceAddr(msg.sender, "IParticipant");
+    if (participantProxy != 0) {
       //todo jaycen can we accomplish the same thing using 820? by defining IParticipantRegistry in the 820reg?
-      require(IParticipant(recipientImplementation).getParticipantRegistry() == getParticipantRegistry());
-    } else if (onlyParticipantCallers == true) {
-      emit InsufficientPermission(
-        msg.sender,
-        _operatorData,
-        _value,
-        _misc
+      require(
+        IParticipant(participantProxy).getParticipantRegistry() == getParticipantRegistry(),
+        "When using a participant proxy, you can only call the mint function with a participant defined in the current ParticipantRegistry contract"
       );
-      revert("Only a participant proxy can call this function right now.");
+    } else if (onlyParticipantCallers == true) {
+        emit InsufficientPermission(
+          msg.sender,
+          _operatorData,
+          _value,
+          _misc
+        );
+        revert("Only a supplier participant proxy can mint when 'onlyParticipantCallers' is true");
```

Comment:
There is no logic changed here, just added revert reasons. Hoping someday soon we get support in truffle ,etc to easily parse these reasons!!!

Reply:
tl;dr  this doesnt impact current implementation of graphql/ui logic, but it may in the future.

Also, this logic will be rewritten soon (actually just copied from `split`) to do this part a bit differently, however, there are still a few tests remaining which rely on the non upgradeable contracts which have no registry assigned. Once all tests are written in a way similar to the new `behaviors/FifoTokenizedCommodityMarket.js`,  that will be applied. This won't change the actual mint logic, and will only apply the final permission logic.

Note: the above statement should be agnostic to the current UI/graphql implementation. However, once we change permissions to use function signatures instead of interfaces, it might have an impact.